### PR TITLE
Add oversampling configuration

### PR DESCRIPTION
**Pull request by zachwe**

### CODE REVIEW

File: uberduck_ml_dev/data_loader.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # AUTOGENERATED! DO NOT EDIT! File to edit: nbs/data_loader.ipynb (unless otherwise specified).
 
-__all__ = ['TextMelDataset', 'TextMelCollate', 'TextAudioSpeakerLoader', 'TextAudioSpeakerCollate',
```
Comment (zachwe) [inline uberduck_ml_dev/data_loader.py:3]: all the significant new code here is in this file

File: uberduck_ml_dev/trainer/base.py [hunk 2]
```diff
@@ -150,9 +150,9 @@ def forward(self, model_output: List, target: List):
 
         mel_loss = mel_loss_batch.mean()
 
-        gate_loss_batch = nn.BCEWithLogitsLoss(pos_weight=None, reduce=False)(
-            gate_out, gate_target
-        ).mean(axis=[1])
+        gate_loss_batch = nn.BCEWithLogitsLoss(
+            pos_weight=self.pos_weight, reduce=False
```
Comment (zachwe) [inline uberduck_ml_dev/trainer/base.py:154]: Bugfix, we weren't plumbing `self.pos_weight` through.

**Review by sjkoelle — COMMENTED** [review]
Seems reasonable.  Does this not sample with replacement?  Probably not a huge deal.  Eventually, we should add undersampling too.

File: uberduck_ml_dev/trainer/base.py (hunk 2 above)
Comment (sjkoelle) [reply uberduck_ml_dev/trainer/base.py:154]: ahhhhhhh.  good catch.  will pull this.

**Comment by zachwe** [comment]
this just duplicates all data for a speaker `weight` times.

I looked into using `imblearn` but it seemed like overkill for what we needed right now, especially since it'd require some massaging to get the data into a proper format.

**Pull request merged**